framework/src/modules/chain/init_steps/attach_swagger.js

```diff
@@ -0,0 +1,12 @@
+const util = require('util');
+const httpApi = require('../helpers/http_api.js');
+
+const promisifiedBootstrapSwagger = util.promisify(httpApi.bootstrapSwagger);
+
+module.exports = async scope =>
+	promisifiedBootstrapSwagger(
+		scope.network.app,
+		scope.config,
+		scope.components.logger,
+		scope
+	);
```

Comment:
I don't know if it is scope of this PR but I really do not like passing this `scope` variable around the modules. We already had a lot of issues with passing and object and reaching its nested properties. It makes mocking really painful, since I have to create a mock object like below for testing this module:

```js
const scope = {
 network : {
  app: sinon.spy(),
 },
config: {..},
components: {
  logger: sinon.stub()
 },
};
```

Also I really don't understand why we are passing child properties as first 3 arguments to `promisifiedBootstrapSwagger`, since we are again already passing the scope object as the 4th argument to this function.

Again, I can understand if in this PR focus wasn't improving the code. But just wanted to raise my concern on this. If possible, please refactor this in this PR or I can create an issue to refactor them later.

Reply:
**Scope**
Would be nice not to have it at all and pass the dependencies directly into logic and modules. But it demands many change to the exisitng structure which is already multiplied in the test application loader. Not the priority for now, but something to carefully refactor later good planning. There is a second aspect to that - modyifying the variables and subscribing to those changes. Right now there is a system component that works de facto like a state container - we should come up with a pattern we could re-use (like React `setState`). We will discuss it further but not on this PR.